src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -413,7 +413,7 @@ public virtual HtmlString GenerateAntiforgery([NotNull] ViewContext viewContext)
             var fullName = GetFullHtmlFieldName(viewContext, expression);
             if (string.IsNullOrEmpty(fullName))
             {
-                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(expression));
+                throw new ArgumentException(Resources.HtmlGenerator_FieldNameCannotBeNullOrEmpty);
```

Comment:
Do not remove the `nameof(expression)` argument from any of these `ArgumentException` constructor calls. `expression` remains the parameter with the problem.